src/hotspot/share/prims/jvmtiEnv.cpp

```diff
@@ -706,7 +706,7 @@ JvmtiEnv::AddToSystemClassLoaderSearch(const char* segment) {
     ObjectLocker ol(loader, THREAD);
 
     // need the path as java.lang.String
-    Handle path = java_lang_String::create_from_platform_dependent_str(segment, THREAD);
+    Handle path = java_lang_String::create_from_platform_dependent_str(THREAD, segment);
```

Comment:
This change will need reverting as an exception is possible as previously noted.

But note that if there was no possibility of exception here then the following check of HAS_PENDING_EXCEPTION should also have been removed.

Reply:
Reverted.